app/src/main/java/ch/epfl/sweng/erpa/services/DataService.java

```diff
@@ -0,0 +1,13 @@
+package ch.epfl.sweng.erpa.services;
+
+import com.annimon.stream.Optional;
+
+import java.util.Set;
+
+import ch.epfl.sweng.erpa.model.UidObject;
+
+public interface DataService<T extends UidObject> {
```

Comment:
Since you want to use a repository pattern, what about making this an abstract class and broke `DataService<T>` for every serializable class `T` and publish the instances on the scope in the DummyRemoteServicesProvider constructor?